cert/cert.go

```diff
@@ -45,16 +45,15 @@ const (
 	duration365d = time.Hour * 24 * 365
 )
 
-var (
-	ErrStaticCert = errors.New("cannot renew static certificate")
-)
+var ErrStaticCert = errors.New("cannot renew static certificate")
 
-// Config contains the basic fields required for creating a certificate
+// Config contains the basic fields required for creating a certificate.
 type Config struct {
-	CommonName   string
-	Organization []string
-	AltNames     AltNames
-	Usages       []x509.ExtKeyUsage
+	CommonName       string
+	Organization     []string
+	AltNames         AltNames
+	Usages           []x509.ExtKeyUsage
+	ExpirationInDays int
```

Comment:
Would it be worth just making this a `*time.Duration` so that it could also go shorter than a day, if the use case called for it?

Reply:
I'd be open to it if the use case ever presents itself.

Reply:
I'd vote for doing so now. I know that users have asked to be able to (for example) set expiration to 10 years; if we make this a duration then we just have to add a DurationFlag to K3s/RKE2 and users can set whatever specific hours/days/years they want without having to convert to an integer number of days.